rtnl/route.go

```diff
@@ -32,8 +43,36 @@ func (c *Conn) RouteAdd(ifc *net.Interface, dst net.IPNet, gw net.IP) (err error
 		Protocol:   unix.RTPROT_BOOT,
 		Type:       unix.RTN_UNICAST,
 		Scope:      uint8(scope),
-		DstLength:  uint8(prefixlen),
+		DstLength:  uint8(dstlen),
+		SrcLength:  uint8(srclen),
 		Attributes: attr,
 	}
 	return c.Conn.Route.Add(tx)
 }
+
+// RouteDel deletes the route to the given destination.
+func (c *Conn) RouteDel(ifc *net.Interface, dst net.IPNet) error {
+	af, err := addrFamily(dst.IP)
+	if err != nil {
+		return err
+	}
+	prefixlen, _ := dst.Mask.Size()
+	scope := addrScope(dst.IP)
+	if len(dst.IP) == net.IPv6len && dst.IP.To4() == nil {
+		scope = unix.RT_SCOPE_UNIVERSE
+	}
+	attr := rtnetlink.RouteAttributes{
+		Dst:      dst.IP,
+		OutIface: uint32(ifc.Index),
+	}
+	tx := &rtnetlink.RouteMessage{
+		Family:     uint8(af),
+		Table:      unix.RT_TABLE_MAIN,
+		Protocol:   unix.RTPROT_BOOT,
```

Comment:
I think we can leave out the proto, type and scope completely. They cannot be different for the same destination.

Reply:
I saw this only now - I can remove it in another PR

Reply:
Nah thats ok. The scope and proto need some work anyway. To see if we can autodetect it.

Reply:
Addressed in https://github.com/jsimonetti/rtnetlink/pull/50